<launch>
    <!-- UAV基本参数 -->
    <arg name="uav_id" default="1"/>
    <arg name="uav_name" default="uav"/>
    <arg name="sim_mode" default="false"/>

    <!-- 搜索模式参数 -->
    <!--
        search_mode: 搜索模式选择
        0 = CIRCLE_CURRENT: 以当前位置为圆心画圆搜索
        1 = CIRCLE_FIXED:   以指定坐标(center_x, center_y)为圆心画圆搜索
        2 = SPIRAL:         螺旋搜索（从中心向外扩展）
        3 = GRID:           栅格搜索（蛇形来回扫描）
    -->
    <arg name="search_mode" default="0"/>

    <!-- 搜索参数 -->
    <arg name="search_height" default="2.5"/>
    <arg name="search_radius" default="2.0"/>
    <arg name="search_speed" default="0.5"/>
    <arg name="search_circles" default="7"/>

    <!-- 指定圆心坐标 (仅在 search_mode=1 或 search_mode=3 时生效) -->
    <arg name="center_x" default="-2.0"/>
    <arg name="center_y" default="-2.0"/>

    <!-- 二维码检测参数 (仿真模式) -->
    <arg name="camera_name" default="/uav$(arg uav_id)/monocular_down"/>
    <arg name="image_topic" default="image_raw"/>
    <arg name="queue_size" default="1" />

    <!-- 启动二维码检测节点 (仿真模式下) -->
    <group if="$(arg sim_mode)">
        <node pkg="landmark_detection_ros" type="landmark_detection_ros_node" name="landmark_detection_ros_node" output="screen">
            <remap from="image_rect" to="$(arg camera_name)/$(arg image_topic)" />
            <remap from="camera_info" to="$(arg camera_name)/camera_info" />

            <param name="queue_size" type="int" value="$(arg queue_size)" />

            <param name="uav_id" type="int" value="1" />
            <param name="x_bias" type="double" value="0" />
            <param name="y_bias" type="double" value="0" />
            <param name="draw_tag_detections_image" type="bool" value="true" />

            <param name="rotate" type="int" value="1" /> <!-- 0: no rotation, 1: rotate 180 degree clockwise, 2: rotate 90 degree clockwise  3: rotate -90 degree clockwise-->

            <param name="algorithm_parameters" type="str" value="$(find sunray_tutorial)/config/aruco_detector_params.json" />
            <param name="camera_parameters" type="str" value="$(find sunray_tutorial)/config/head_camera_sim.yaml" />
            <param name="local_saving_path" type="str" value="$(find sunray_tutorial)/output" />
            
        </node>
    </group>

    <!-- 启动搜索目标节点 -->
    <node pkg="sunray_tutorial" type="search_target_and_send_to_ugv" name="search_target_and_send_to_ugv_$(arg uav_id)" output="screen">
        <param name="uav_id" value="$(arg uav_id)"/>
        <param name="uav_name" value="$(arg uav_name)"/>
        <param name="search_mode" value="$(arg search_mode)"/>
        <param name="search_height" value="$(arg search_height)"/>
        <param name="search_radius" value="$(arg search_radius)"/>
        <param name="search_speed" value="$(arg search_speed)"/>
        <param name="search_circles" value="$(arg search_circles)"/>
        <param name="center_x" value="$(arg center_x)"/>
        <param name="center_y" value="$(arg center_y)"/>
    </node>
</launch>
